pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -30,8 +30,7 @@
   isPy3k, pythonOlder }:
 
 # assert that everything needed for cuda is present and that the correct cuda versions are used
-assert !cudaSupport || (let majorIs = lib.versions.major cudatoolkit.version;
-                        in majorIs == "9" || majorIs == "10" || majorIs == "11");
+assert !cudaSupport || lib.versionAtLeast cudatoolkit.version "10.2.0";
```

Comment:
The binary uses cuda11.3.

Reply:
Thanks cuda that is passed in should probably be 11.3 by default then. 

My take that 10.2.0 is the minimum was based on a CMake error I got, but I haven't properly investigated if that's truly the lower bound.
```
-- Found CUDA: /nix/store/rclmj9izhxwhgwrgjk97mka3ndm4bzyn-cudatoolkit-10.1.243 (found version "10.1") 
-- Caffe2: CUDA detected: 10.1
-- Caffe2: CUDA nvcc is: /nix/store/rclmj9izhxwhgwrgjk97mka3ndm4bzyn-cudatoolkit-10.1.243/bin/nvcc
-- Caffe2: CUDA toolkit directory: /nix/store/rclmj9izhxwhgwrgjk97mka3ndm4bzyn-cudatoolkit-10.1.243
CMake Error at cmake/public/cuda.cmake:42 (message):
  PyTorch requires CUDA 10.2 or above.
Call Stack (most recent call first):
  cmake/Dependencies.cmake:1191 (include)
  CMakeLists.txt:653 (include)
```


(Btw this check doesn't seem to trigger for cudnn_cudatoolkit which is currently against 10.1.0 - I'm looking at fixing via https://github.com/NixOS/nixpkgs/pull/164820)

Reply:
The problem was that I was a little misunderstood. I think this is okay.

Reply:
To unblock you on the CUDA/cuDNN issue as quickly as possible, I would suggest just overriding with the specific versions you need here

https://github.com/NixOS/nixpkgs/blob/83e8ce412b4c7561b28acc924bd46a4bd08c2351/pkgs/top-level/python-packages.nix#L8312-L8314

a la

https://github.com/NixOS/nixpkgs/blob/83e8ce412b4c7561b28acc924bd46a4bd08c2351/pkgs/top-level/python-packages.nix#L4203-L4215

There's no need to rely too heavily on the `cudatoolkit` and `cudnn` aliases if they aren't currently pointing to the versions that you need.

Reply:
I've removed pytorch source build from this PR now since it's already merged elsewhere :+1: